docs/content/docs/connectors/datastream/pulsar.md

```diff
@@ -0,0 +1,413 @@
+// Pulsar connector would consume from the latest available message if the message id doesn't exist.
+// The start message is included in consuming result. 
+StartCursor.fromMessageId(MessageId)
+
+// Include or exclude the start message by using the second boolean parameter.
+StartCursor.fromMessageId(MessageId, boolean)
+
+// Start from the specified message time (Message<byte[]>#getEventTime()).
+StartCursor.fromMessageTime(long)
+```
+
+{{< hint info >}}
+Each Pulsar message belongs to an ordered sequence on its topic.
+The sequence ID (```MessageId```) of the message is ordered in that sequence.
+MessageId contains some extra information on how the message was stored (the ledger, entry, partition),
+you can create a MessageId by using ```DefaultImplementation.newMessageId(long ledgerId, long entryId, int partitionIndex)```
+{{< /hint >}}
+
+### Boundedness
+
+Pulsar source is designed to support both streaming and batch running mode.
+By default, the PulsarSource is set to run in streaming manner, thus never stops until Flink job fails or is cancelled.
+You can use ```setBoundedStopCursor(StopCursor)``` to specify stopping position and set the source running in batch mode.
+When all partitions have reached their stopping position, the source will exit.
+
+You can also set PulsarSource running in streaming mode, but still stop at the stopping position by using ```setUnboundedStopCursor(StopCursor)```.
+The source will exit when all partitions reach their specified stopping position.
+
+Built-in stop cursor include:
+
+```java
+// Connector will never stop consuming
+StopCursor.never()
+
+// Stop at the latest available message in Pulsar when connector start consuming.
+StopCursor.latest()
+
+// Stop when connector meet a given message, or stop at a message which is produced after this given message.
+StopCursor.atMessageId(MessageId)
+
+// Stop but include the given message in consuming result.
+StopCursor.afterMessageId(MessageId)
+
+// Stop at the specified message time (Message<byte[]>#getEventTime()).
+StopCursor.atEventTime(long)
+```
+
+### Configurable Options
+
+In addition to config options described above, you can set arbitrary options for PulsarSource, PulsarClient,
+PulsarAdmin and Pulsar Consumer by using ```setConfig(ConfigOption<T>, T)``` and ```setConfig(Configuration)```.
+
+#### Pulsar Client Options
+
+Pulsar connector use both [client API](https://pulsar.apache.org/docs/en/client-libraries-java/)
+for creating the consuming instance. Pulsar connector extract most parts of Pulsar's ```ClientConfigurationData```,
+which is required for creating a Pulsar client, as Flink config options in ```PulsarOptions```.
+
+{{< generated/pulsar_client_configuration >}} 
+
+#### Pulsar Admin Options
+
+[Admin API](https://pulsar.apache.org/docs/en/admin-api-overview/) is used for querying topic metadata
+and filter the desired topics when connector use topic pattern subscription. It would share most part of the
```

Comment:
```suggestion
[Admin API](https://pulsar.apache.org/docs/en/admin-api-overview/) is used for querying topic metadata
and for discovering the desired topics when connector use topic pattern subscription. It would share most part of the
```